wallet/src/Pos/Wallet/Web/State/Storage.hs

```diff
@@ -328,27 +428,34 @@ removeCustomAddress t (addr, hh) = do
         customAddressL t . at addr .= Nothing
     return exists
 
+-- | Add given account to wallet db.
 createAccount :: AccountId -> CAccountMeta -> Update ()
 createAccount accId cAccMeta =
     wsAccountInfos . at accId %= Just . fromMaybe (AccountInfo cAccMeta mempty mempty 0)
 
--- `isReady` is marked False when an additional step such as syncing is still needed.
+-- | Create a new wallet with given parameters.
+-- @isReady@ should be set to @False@ when syncing is still needed.
 createWallet :: CId Wal -> CWalletMeta -> Bool -> POSIXTime -> Update ()
 createWallet cWalId cWalMeta isReady curTime = do
     let info = WalletInfo cWalMeta curTime curTime NotSynced mempty isReady
     wsWalletInfos . at cWalId %= (<|> Just info)
 
+-- | Add new address given 'CWAddressMeta' (which contains information about
+-- target wallet and account too).
 addWAddress :: CWAddressMeta -> Update ()
 addWAddress addrMeta@CWAddressMeta{..} = do
     let accInfo :: Traversal' WalletStorage AccountInfo
         accInfo = wsAccountInfos . ix (addrMetaToAccount addrMeta)
     whenJustM (preuse accInfo) $ \info -> do
         let mAddr = info ^. aiAddresses . at cwamId
         when (isNothing mAddr) $ do
+            -- Here we increment current account's last address index
+            -- and assign its value to sorting index of newly created address.
             accInfo . aiUnusedKey += 1
             let key = info ^. aiUnusedKey
             accInfo . aiAddresses . at cwamId ?= AddressInfo addrMeta key
 
+-- | TODO: what happens here?...
```

Comment:
Should this be addressed?

Reply:
Yes. I'm genuinely confused about this function.

Reply:
Indeed, it's not used anywhere :peka: